### PR TITLE
Refresh cart summary when adding a voucher

### LINKED ISSUE #17290: Cart not refreshed when adding voucher with a gift product
**Describe the bug**
If you create a voucher with a free gift, when you add it in the checkout page the cart is not refreshed.

**To Reproduce**
1. Create a cart rule with a gift product and with a coupon code

image

image

2. Add a product to the cart
3. Go to the checkout (not the cart summary)
4. Add the coupon code

image
image

image

5. After refreshing the page

image

6. If you delete the voucher, the cart is not updated neither. The gift is not removed:

image

image

**Additional information**
PrestaShop version: 1.7.6.3

### PR DESCRIPTION
**Pull request by idnovate**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Refresh cart summary when adding a voucher
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #17290
| How to test?  | Check bug description

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17294)

### CODE REVIEW

**Comment by Progi1984** [comment]
@idnovate Thank you for your contribution. We have a problem with this PR : he doesn't pass sanity test on Travis on PHP 7.2  (https://travis-ci.com/PrestaShop/PrestaShop/jobs/278305360#L798).

**Comment by Progi1984** [comment]
@idnovate I rebase your PR with the latest version of develop. Now it pass Travis ;)

**Review by Progi1984 — DISMISSED** [review]

File: themes/classic/templates/checkout/_partials/cart-summary-products.tpl [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ *}
+
+ <div class="cart-summary-products">
+    <p>{$cart.summary_string}</p>
```
Comment (PierreRambaud) [inline themes/classic/templates/checkout/_partials/cart-summary-products.tpl]: use two spaces for indentation please

File: themes/classic/templates/checkout/_partials/cart-summary-products.tpl [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            {l s='show details' d='Shop.Theme.Actions'}
+            <i class="material-icons">expand_more</i>
+        </a>
+    </p>
+
+    {block name='cart_summary_product_list'}
+    <div class="collapse" id="cart-summary-product-list">
+        <ul class="media-list">
+            {foreach from=$cart.products item=product}
+            <li class="media">{include file='checkout/_partials/cart-summary-product-line.tpl' product=$product}</li>
```
Comment (PierreRambaud) [inline themes/classic/templates/checkout/_partials/cart-summary-products.tpl]: As I said for #17296 wrong indentation in this file.

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Problem with indentation

**Review by Progi1984 — DISMISSED** [review]

File: themes/classic/templates/checkout/_partials/cart-summary-products.tpl [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    </a>
+  </p>
+
+  {block name='cart_summary_product_list'}
+    <div class="collapse" id="cart-summary-product-list">
+      <ul class="media-list">
+        {foreach from=$cart.products item=product}
+          <li class="media">{include file='checkout/_partials/cart-summary-product-line.tpl' product=$product}</li>
+        {/foreach}
+     </ul>
```
Comment (PierreRambaud) [inline themes/classic/templates/checkout/_partials/cart-summary-products.tpl]:
```suggestion
      </ul>
```

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @idnovate,

Issue not fixed using your PR
I attached a screen record
https://drive.google.com/file/d/1XpjvTSzwTqUSsdvjnVCqvfHZND27zmJC/view
Thanks to check & feedback.

**Comment by idnovate** [comment]
> Hi @idnovate,
> 
> Issue not fixed using your PR
> I attached a screen record
> https://drive.google.com/file/d/1XpjvTSzwTqUSsdvjnVCqvfHZND27zmJC/view
> Thanks to check & feedback.

- Have you cleared cache and force compilation?
- Could you please check if you have any JS error?
- Could you please check if you have field `cart_summary_products` in the response, and if it's not empty?

Thanks!

**Comment by khouloudbelguith** [comment]
@idnovate,
>Have you cleared cache and force compilation?

Yes!
>Could you please check if you have any JS error?

No, there is no error.
I attached a screen record
https://drive.google.com/file/d/1enqLAv1RIBOuBNqMacLM7s0_DalRuId9/view

Thanks!

**Comment by idnovate** [comment]
> @idnovate,
> 
> > Have you cleared cache and force compilation?
> 
> Yes!
> 
> > Could you please check if you have any JS error?
> 
> No, there is no error.
> I attached a screen record
> https://drive.google.com/file/d/1enqLAv1RIBOuBNqMacLM7s0_DalRuId9/view
> 
> Thanks!

Great, and for the last point, could you please record "Network" tab and check the response?

**Comment by khouloudbelguith** [comment]
@idnovate, here's the screen record
https://drive.google.com/file/d/18lp37Eqcc4W45lCk-OiAvB5D4XHtq329/view

Thanks!

**Comment by idnovate** [comment]
> @idnovate, here's the screen record
> https://drive.google.com/file/d/18lp37Eqcc4W45lCk-OiAvB5D4XHtq329/view
> 
> Thanks!

Thanks for helping me to debug why it's not working for you!

The response is OK. The issue seems to be then in file `themes/_core/js/cart.js`. Could you please check if you have this line?

`$('.cart-summary-products').replaceWith(resp.cart_summary_products);`

Please note that you need to generate the final JS, I don't know how to do it! :sweat: To test it I directly edit the generated JS.

**Comment by khouloudbelguith** [comment]
Hi @idnovate,

Yes!
image

Thanks!

**Comment by idnovate** [comment]
Then the problem is this:

> Please note that you need to generate the final JS, I don't know how to do it! 😓 To test it I directly edit the generated JS.

This JS is not added in the front, but `/themes/core.js` that include this JS. With webpack?

**Comment by khouloudbelguith** [comment]
@idnovate, I will check & feedback.

Thanks!

**Comment by khouloudbelguith** [comment]
Hi @idnovate,

It is OK!
https://drive.google.com/file/d/1VJtKiVVpqSqPa4gDyWc9dX_SCb62_tJv/view

Thanks!

**Comment by matks** [comment]
Thank you @idnovate

**Pull request merged**